book/src/pil/declarations.md

```diff
@@ -12,12 +12,14 @@ This syntax can be used for constants, fixed columns, witness columns and even (
 functions that can transform expressions. The kind of symbol is deduced by its type and the
 way the symbol is used:
 
-- Symbols without a value are witness columns. Their type can be omitted. If it is given, it must be ``int -> fe`` or its shorthand ``col``.
+- Symbols without a value are witness columns or arrays of witness columns. Their type can be omitted. If it is given, it must be ``col`` or ``col[]``.
 - Symbols evaluating to a number or with type ``fe`` are constants.
 - Symbols without type but with a value that is a function with a single parameter are fixed columns.
-- Symbols defined with a value and type ``int -> fe`` or its shorthand ``col`` are also fixed columns.
+- Symbols defined with a value and type ``col`` (or ``col[]``) are fixed columns (or arrays of fixed columns).
 - Everything else is a "generic symbol" that is not a column or constant.
 
+> Note that the type ``col`` is the same as ``int -> fe``, so ``let w: int -> fe`` also declares a witness column.
```

Comment:
`col` and `int->fe` are identical types. You could theoretically have a utility function with that signature, but if you don't reference it in an expression context, it will be optimized away (or at least I hope), so I think that's fine.

Reply:
So, if I do
```
    let foo: int -> fe = |i| i * i;
    let bar: col = |i| foo(i + (1 << 30));
```

After analysis and optimization this becomes:
```
    col fixed foo(i) { (i * i) };
    col fixed bar(i) { Fibonacci.foo((i + (1 << 30))) };
```

But I never wanted a fixed column `foo` and I'm not even interested in the values (because they are only for $0 \le i < degree$).

Reply:
I guess you could say any `int -> fe` *that is referenced in an identity* should become a fixed column? In this case, we could tweak the optimizer to remove `foo` before it is materialized. I think this is what you meant, right?

Reply:
Yes, that's what I meant.

Reply:
-> https://github.com/powdr-labs/powdr/issues/1029

Reply:
OK, but what if I want to use `foo` in a hint? It shouldn't be removed but also not be a column?